src/OpenTelemetry/Sdk.cs

```diff
@@ -163,11 +165,29 @@ public static TracerProvider CreateTracerProvider(Action<TracerProviderBuilder>
 
         public static TracerProvider CreateTracerProvider(IEnumerable<string> sources, Sampler sampler = null, Resource resource = null)
         {
-            var activitySources = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+            if (sources == null)
+            {
+                throw new ArgumentNullException(nameof(sources));
+            }
+
+            if (!sources.Any())
+            {
+                throw new ArgumentException($"{nameof(sources)} collection is empty.");
+            }
+
+            var wildcardMode = false;
 
             foreach (var name in sources)
             {
-                activitySources[name] = true;
+                if (string.IsNullOrWhiteSpace(name))
```

Comment:
@cijothomas Do we still need to listen to the blank source for the legacy Activity objects? Or did the ActivitySourceAdapter take care of that?